### PR TITLE
feat: NTRN-333. Add treasury contract documentation

### PR DESCRIPTION
**Pull request by albertandrejev**
We need to write an extensive, thorough documentation section for this contract. The purpose of the contract, its interface and interactions with other contracts must be well-documented.

### CODE REVIEW

File: docs/neutron/treasury/messages.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Messages
+
+## MsgTransfer
+
+```protobuf
```
Comment (foxpy) [inline docs/neutron/treasury/messages.md:5]:
```suggestion
```rust
```

File: docs/neutron/treasury/state.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
+# State
+
+```protobuf
```
Comment (foxpy) [inline docs/neutron/treasury/state.md:3]:
```suggestion
```rust
```

**Review by foxpy — CHANGES_REQUESTED** [review]
Also this PR is expected to be targeted to `scratchpad/treasury-distribution-reserve` branch.

File: docs/neutron/treasury/messages.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
+# Messages
+
+## MsgTransfer
```
Comment (foxpy) [inline docs/neutron/treasury/messages.md:3]: `MsgTransfer`???

File: docs/neutron/treasury/state.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    pub security_dao_address: Addr,
+
+    // Denomintator used int the vesting release function
+    pub vesting_denominator: u128,
+}
+```
+
+## Stats
+
+Returns contract coins distrivution stats. Return has a following schema:
```
Comment (foxpy) [inline docs/neutron/treasury/state.md:44]:
```suggestion
Returns contract coins distribution stats. Return has a following schema:
```

File: docs/neutron/treasury/messages.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+Transfer the contract's ownership to another account. Can be executed by `main_dao_address` only.
+
+
+## Distribute
+Distribute pending funds between Bank and Distribution accounts. Can be executed by any address, but not more than `min_period` of heights between calls.
+
+## UpdateConfig
+
+Update treasury contract configuration. Permissioned can be executed only by Main DAO.
```
Comment (foxpy) [inline docs/neutron/treasury/messages.md:40]:
```suggestion
Update treasury contract configuration. Permissioned, can be executed only by Main DAO.
```

File: docs/neutron/treasury/overview.md [hunk 6]
```diff
@@ -0,0 +1,9 @@
+# Overview
+
+This document describes the treasury contract for the Neutron network.
+
+**The Treasury** contract holds the vested NTRNs and sends them to the Reserve and Distribution contracts. Contract is owned by DAO and is instantiated in Genesis and responsible of the first step of tokens distribution. Treasury tokens are vested based on on-chain activity: the more NTRN tokens are burned while processing block fees (see above), the more tokens get unlocked in the treasury; 
+
+Main approach in the released coins calculation based on the following expression: `burnt_tokens * a_multiplier`. The `a_multiplier` is a linear function: fn(y) = x / `<configurable denominator>`. Configurable denominator should be set based on coins burning rate and preffered duration. Take into account that while initially, one burnt tokens equals multiple NTRN tokens made liquid, the flow of new tokens into the Treasury progressively slows down until the tokens supply is exhausted and the tokenomy becomes deflationary.
```
Comment (foxpy) [inline docs/neutron/treasury/overview.md:7]:
```suggestion
Main approach in the released coins calculation based on the following expression: `burnt_tokens * a_multiplier`. The `a_multiplier` is a linear function: fn(y) = x / `<configurable denominator>`. Configurable denominator should be set based on coins burning rate and preferred duration. Take into account that while initially, one burnt tokens equals multiple NTRN tokens made liquid, the flow of new tokens into the Treasury progressively slows down until the tokens supply is exhausted and the tokenomy becomes deflationary.
```

**Review by foxpy — CHANGES_REQUESTED** [review]

**Review by foxpy — APPROVED** [review]

File: docs/neutron/treasury/messages.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+}
+```
+
+## TransferOwnership 
+
+Transfer the contract's ownership to another account. Can be executed by `main_dao_address` only.
+
+
+## Distribute
+Distribute pending funds between Bank and Distribution accounts. Can be executed by any address, but not more than `min_period` of heights between calls.
```
Comment (NeverHappened) [inline docs/neutron/treasury/messages.md:36]: space missed

File: docs/neutron/treasury/messages.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+## UpdateConfig
+
+Update treasury contract configuration. Permissioned, can be executed only by Main DAO.
+
+```rust
+UpdateConfig {
+    /// Distribution rate (0-1) which goes to distribution contract
```
Comment (NeverHappened) [inline docs/neutron/treasury/messages.md:44]: how about using math notation to show inclusion? `[0; 1]`

File: docs/neutron/treasury/messages.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+    distribution_rate: Option<Decimal>,
+
+    /// Minimum period between distribution calls in amount of blocks
+    min_period: Option<u64>,
+
+    /// Address of distribution contract, which will receive funds defined but distribution_rate %
```
Comment (NeverHappened) [inline docs/neutron/treasury/messages.md:50]: typo `but` -> `by`

Comment (NeverHappened) [reply docs/neutron/treasury/messages.md:50]: and probably no need for `,`

File: docs/neutron/treasury/messages.md [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+    distribution_contract: Option<String>,
+
+    /// Address of reserve contract, which will receive funds defined by 100-distribution_rate %
+    reserve_contract: Option<String>,
+
+    /// Address of the security DAO contract
+    security_dao_address: Option<String>,
+
+    /// Denomintator used in the vesting release function
```
Comment (NeverHappened) [inline docs/neutron/treasury/messages.md:59]: typo -> `Denominator`

File: docs/neutron/treasury/messages.md [hunk 11]
```diff
@@ -0,0 +1,69 @@
+# Messages
+
+```rust
+#[pausable]
+pub enum ExecuteMsg {
+    /// Transfer the contract's ownership to another account
```
Comment (NeverHappened) [inline docs/neutron/treasury/messages.md:6]: [permissioned - executable only by Main DAO]

File: docs/neutron/treasury/state.md [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+pub enum QueryMsg {
+    /// The contract's configurations; returns [`ConfigResponse`]
+    Config {},
+    Stats {},
+}
+```
+
+## Config
+
+Returns current treasure contract configuration. Return has a following schema:
```
Comment (NeverHappened) [inline docs/neutron/treasury/state.md:14]: typo -> `Treasury`

Comment (NeverHappened) [reply docs/neutron/treasury/state.md:14]: maybe better - `Returns an object with following schema:`

File: docs/neutron/treasury/state.md [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+
+
+```rust
+pub struct Config {
```
Comment (NeverHappened) [inline docs/neutron/treasury/state.md:18]: in some places there are spaces between fields, in others not, suggest unifying codestyle

File: docs/neutron/treasury/overview.md [hunk 14]
```diff
@@ -0,0 +1,9 @@
+# Overview
+
+This document describes the treasury contract for the Neutron network.
+
+**The Treasury** contract holds the vested NTRNs and sends them to the Reserve and Distribution contracts. Contract is owned by DAO and is instantiated in Genesis and responsible of the first step of tokens distribution. Treasury tokens are vested based on on-chain activity: the more NTRN tokens are burned while processing block fees (see above), the more tokens get unlocked in the treasury;
```
Comment (NeverHappened) [inline docs/neutron/treasury/overview.md:5]: two `and`'s in one sentence -> can be solved with comma)

Comment (NeverHappened) [reply docs/neutron/treasury/overview.md:5]: Maybe `the more tokens distributed from the Treasury`? Not sure tho, but it seems clearer that we distribute tokens from treasury

File: docs/neutron/treasury/overview.md [hunk 15]
```diff
@@ -0,0 +1,9 @@
+# Overview
+
+This document describes the treasury contract for the Neutron network.
+
+**The Treasury** contract holds the vested NTRNs and sends them to the Reserve and Distribution contracts. Contract is owned by DAO and is instantiated in Genesis and responsible of the first step of tokens distribution. Treasury tokens are vested based on on-chain activity: the more NTRN tokens are burned while processing block fees (see above), the more tokens get unlocked in the treasury; 
+
+Main approach in the released coins calculation based on the following expression: `burnt_tokens * a_multiplier`. The `a_multiplier` is a linear function: fn(y) = x / `<configurable denominator>`. Configurable denominator should be set based on coins burning rate and preferred duration. Take into account that while initially, one burnt tokens equals multiple NTRN tokens made liquid, the flow of new tokens into the Treasury progressively slows down until the tokens supply is exhausted and the tokenomy becomes deflationary.
```
Comment (NeverHappened) [inline docs/neutron/treasury/overview.md:7]: Maybe rephrase a bit?
`To calculate how many coins we distribute (release) we use following expression: ....`

Comment (NeverHappened) [reply docs/neutron/treasury/overview.md:7]: maybe clarify a bit: `and preferred total duration of vesting`?

Comment (NeverHappened) [reply docs/neutron/treasury/overview.md:7]: `while initially, one burnt ` here don't need comma

File: docs/neutron/treasury/overview.md [hunk 16]
```diff
@@ -0,0 +1,9 @@
...
+
+Treasury contract can be configured only by DAO. The contract has the `min_period` parameter, and the `distribute` method can not be called more than once every `min_period`. The `distribute` call is permissionless and can be called by anybody.
```
Comment (NeverHappened) [inline docs/neutron/treasury/overview.md:9]: `can not` -> `cannot`

File: docs/neutron/treasury/messages.md [hunk 17]
```diff
@@ -0,0 +1,69 @@
...
+    TransferOwnership(String),
+
+    /// Distribute pending funds between Bank and Distribution accounts
+    Distribute {},
```
Comment (NeverHappened) [inline docs/neutron/treasury/messages.md:10]: [permissionless]

File: docs/neutron/treasury/state.md [hunk 18]
```diff
@@ -0,0 +1,72 @@
...
+    /// Denom of the main coin
+    pub denom: String,
+
+    /// Address of the main DAO contract
+    pub main_dao_address: Addr,
+
+    /// Address of the security DAO contract
+    pub security_dao_address: Addr,
+
+    // Denomintator used int the vesting release function
```
Comment (NeverHappened) [inline docs/neutron/treasury/state.md:37]: typos: `Denominator used in`

**Review by NeverHappened — CHANGES_REQUESTED** [review]

File: docs/neutron/treasury/overview.md [hunk 19]
```diff
@@ -0,0 +1,9 @@
+# Overview
+
+This document describes the treasury contract for the Neutron network.
+
+**The Treasury** contract holds the vested NTRNs sends them to the Reserve and Distribution contracts. Contract is owned by DAO and is instantiated in Genesis and responsible of the first step of tokens distribution. Treasury tokens are vested based on on-chain activity: the more NTRN tokens are burned while processing block fees (see above), the more tokens distributed from the Treasury.
```
Comment (NeverHappened) [inline docs/neutron/treasury/overview.md:5]: responsible *for* the first step

File: docs/neutron/treasury/state.md [hunk 20]
```diff
@@ -0,0 +1,75 @@
+# Queries
+
+```rust
+#[pausable_query]
+pub enum QueryMsg {
+    /// The contract's configurations; returns [`ConfigResponse`]
+    Config {},
+    Stats {},
```
Comment (NeverHappened) [inline docs/neutron/treasury/state.md:8]: no comment for the field

**Review by NeverHappened — APPROVED** [review]

File: docs/neutron/treasury/overview.md (hunk 19 above)
Comment (foxpy) [reply docs/neutron/treasury/overview.md]: > **The Treasury** contract holds the vested NTRNs sends them to the Reserve and Distribution contracts.

feels like there is a missing comma.

**Review by NeverHappened — APPROVED** [review]

**Review by foxpy — APPROVED** [review]
